pkg/sql/sem/tree/casts.go

```diff
@@ -434,6 +436,12 @@ func AdjustValueToType(typ *types.T, inVal Datum) (outVal Datum, err error) {
 			} else if _, ok := inVal.(*DCollatedString); ok {
 				return NewDCollatedString(strings.TrimRight(sv, " "), typ.Locale(), &CollationEnvironment{})
 			}
+		} else if typ.Oid() == oid.T_char {
+			if _, ok := AsDString(inVal); ok {
+				return NewDString(util.TruncateString(sv, 1)), nil
```

Comment:
do we have to TruncateString again here?
i think we can omit the TrimRight above as well - oops.

Reply:
thank you, good catch